### ISSUE TITLE
Addressing issues regarding this system

### ISSUE THREAD

**Issue by DonaldTsang** [KichangKim/DeepDanbooru#5]
It would be easy to break such a system and cause mis-tagging by using these libraries as a demonstration on the weakness of using NN for automated image tagging.
- https://github.com/tensorflow/cleverhans
- https://github.com/bethgelab/foolbox
- https://github.com/advboxes/AdvBox

Here are some of my proposals:
1. image augmentation to prevent overfitting https://github.com/aleju/imgaug
2. usage of multiple models for the same task https://arxiv.org/abs/1809.00065
3. de-noising the image using something like Waifu2X and others

I have talked about something similar in https://github.com/halcy/DeepDanbooruActivationMaps/issues/3

**Comment by DonaldTsang** [comment]
Some useful information regarding the segmentation of images

**Comment by DonaldTsang** [comment]
Some ideas in how to implement a Semantic Segmentation dataset
1. The initial dataset phase
- Gather a list of images with strong heatmap confidence
- Use Recaptha's 3x3 5F-3T-1U test to refine the borders (remember to augment and noise them)
- Collect results from volunteers
2. The Semantic Segmentation training phase
- Create the system model (or better yet multiple models)
- Use the collected data to train the system
- Optimize the system speed and accuracy wise regarding ensembles
3. The data refinement phase
- Increase the scope of images used
- Use Recaptha's 3x3 5F-3T-1U test to refine the borders (remember to augment and noise them)
- Use volunteer's results to refine the Semantic Segmentation

**Comment by DonaldTsang** [comment]
I am just going to put this here, for those who wants to go from label to table.
- https://people.csail.mit.edu/mrub/ObjectDiscovery/
- https://www.researchgate.net/publication/325052999_Object_Discovery_and_Cosegmentation_Based_on_Dense_Correspondences
- https://www.researchgate.net/publication/320350075_Unsupervised_Image_Co-segmentation_via_Guidance_of_Simple_Images
